packages/components/src/FigspecViewer/InspectorView/utils.ts

```diff
@@ -0,0 +1,271 @@
+          : `${node.cornerRadius}px`;
+    }
+
+    // colors, background, fill
+    if (node.backgroundColor || node.backgroundColor) {
+      const color = node.backgroundColor || node.background?.[0].color;
+      this.background = extractColorStyle(color);
+    }
+
+    const fillColor = node.fills?.[0];
+    if (fillColor && fillColor.visible !== false) {
+      if (node.type === "TEXT") {
+        this.color = extractColorStyle(fillColor.color);
+      } else if (fillColor.type.includes("GRADIENT")) {
+        this.backgroundImage = extractGradientColorStyle(
+          (fillColor as unknown) as ElementGradientColor
+        );
+      } else if (fillColor.type === "SOLID") {
+        this.background = extractColorStyle(fillColor.color);
+      }
+    }
+
+    // borders
+    if (node.strokes && node.strokes.length > 0) {
+      this.borderColor = extractColorStyle(
+        node.strokes[0].color as ElementColor
+      );
+      this.border = `${node.strokeWeight}px solid ${this.borderColor}`;
+    }
+
+    // box-shadow
+    if (node.effects && node.effects.length > 0) {
+      const { offset, radius, color } = node.effects[0];
+      this.boxShadowColor = extractColorStyle(color as Figma.Color);
+      this.boxShadow = `${offset?.x || 0}px ${offset?.y || 0}px 0 ${radius} ${
+        this.boxShadowColor
+      }`;
+    }
+  }
+
+  getStyles() {
+    return [
+      this.height && { property: "height", value: this.height },
+      this.width && { property: "width", value: this.width },
+      this.fontFamily && { property: "font-family", value: this.fontFamily },
+      this.fontSize && { property: "font-size", value: this.fontSize },
+      this.fontWeight && { property: "font-weight", value: this.fontWeight },
+      this.lineHeight && { property: "line-height", value: this.lineHeight },
+      this.borderRadius && {
+        property: "border-radius",
+        value: this.borderRadius,
+      },
+      this.backgroundImage && {
+        property: "background-image",
+        value: this.backgroundImage,
+      },
+      this.boxShadow && {
+        property: "box-shadow",
+        value: this.boxShadow,
+        color: this.boxShadowColor,
+      },
+      this.border && {
+        property: "border",
+        value: this.border,
+        color: this.borderColor,
+      },
+      this.background && {
+        property: "background",
+        value: this.background,
+        color: this.background,
+      },
+      this.color && { property: "color", value: this.color, color: this.color },
+    ].filter(Boolean) as CSSRule[];
+  }
+
+  getStyleSheet() {
+    return this.getStyles().map(getStyleRule).join("\n");
+  }
+}
+
+const rgbToIntArray = (color: ElementColor) => [
+  Math.trunc(255 * color.r),
+  Math.trunc(255 * color.g),
+  Math.trunc(255 * color.b),
+];
+
+const rgbToHex = (color: ElementColor) => {
+  const [r, g, b] = rgbToIntArray(color);
+  return "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
```

Comment:
Need 0 padding?

Reply:
Not that I know of, `r`, `g` and `b` will always be between 0 and 255, and with those values it will always output from `#000000` to `#ffffff`

Reply:
Ah, sorry! I misunderstood the code :alien: